File: src/code-samples/deepagents/tools.py
"""Deep Agents tools page examples."""

# :snippet-start: tools-pass-tools-py
from deepagents import create_deep_agent

# :remove-start:
def search(query: str) -> str:
    """Search the web."""
    return query


def fetch_url(url: str) -> str:
    """Fetch a URL."""
    return url


def run_query(sql: str) -> str:
    """Run a SQL query."""
    return sql

# :remove-end:

agent = create_deep_agent(
    model="anthropic:claude-sonnet-4-6",
    tools=[search, fetch_url, run_query],
)
# :snippet-end:

# :remove-start:
assert agent is not None
# :remove-end:

File: src/code-samples/deepagents/tools.ts
// :snippet-start: tools-pass-tools-js
import { createDeepAgent } from "deepagents";

// :remove-start:
async function search(query: string): Promise<string> {
  return query;
}

async function fetchUrl(url: string): Promise<string> {
  return url;
}

async function runQuery(sql: string): Promise<string> {
  return sql;
}
// :remove-end:

const agent = await createDeepAgent({
  model: "anthropic:claude-sonnet-4-6",
  tools: [search, fetchUrl, runQuery],
});
// :snippet-end:

// :remove-start:
if (!agent) throw new Error("agent not created");
// :remove-end:
